src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,185 @@
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    public static final int NO_ELEMENT_ARRAY_LENGTH = 0;
+    public static final int MIN_MONTH = 1;
+    public static final int MAX_MONTH = 12;
+    public static final int MAX_DAY_IN_MONTH = 31;
+    public static final String TIME_ZONE = "+02:00";
+    public static final int START_DAY_POSITION = 0;
+    public static final int END_DAY_POSITION = 2;
+    public static final int START_MONTH_POSITION = 4;
+    public static final int END_MONTH_POSITION = 6;
+
+    Locale locale = new Locale("en", "US");
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate date = LocalDate.now();
+        switch (datePart) {
+            case FULL:
+                return String.valueOf(date);
+            case YEAR:
+                return String.valueOf(date.getYear());
+            case MONTH:
+                return String.valueOf(date.getMonth());
+            case DAY:
+                return String.valueOf(date.getDayOfMonth());
+            default:
+                throw new DateTimeException("Can't get date!");
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        if (dateParams == null || dateParams.length == NO_ELEMENT_ARRAY_LENGTH
+                || dateParams[1] < MIN_MONTH || dateParams[1] > MAX_MONTH) {
+            return Optional.empty();
+        }
+        int year = dateParams[0];
+        int month = dateParams[1];
+        int dayOfMonth = dateParams[2];
+        return Optional.ofNullable(LocalDate.of(year, month, dayOfMonth));
```

Comment:
you can try do this with try/catch block

Reply:
cool idea! done